js/services/api/guest-invites.js

```diff
@@ -39,45 +80,63 @@ const processEmailInvites = (potluckId, guestEmails) => {
   });
 };
 
-const createPotluckGuest = (potluckId, uIds) => {
-  const potluckUserGuest = {
-    inviteIssued: true,
-    inviteIssueDate: moment(Date.now()).format('YYYY-MM-DD'),
-    inviteAccepted: false,
-    inviteAcceptedDate: null,
-    inviteCancelled: null,
-    inviteCancelledDate: null,
-  };
+const getPotluckInvites = () => {
+  return firebase.database()
+    .ref()
+    .child('/potLuckInvites')
+    .once('value');
+};
 
-  uIds.forEach((uId) => {
-    const updates = {
-      [`/potLuckGuests/${potluckId}/${uId}`]: potluckUserGuest,
-    };
+const getPotluckInviteEmail = (potluckId, encodedEmail) => {
+  const result = firebase
+    .database()
+    .ref(`/potLuckInvites/${potluckId}/guests/${encodedEmail}`)
+    .once('value');
 
-    return api.change(updates);
-  });
+  if (result) return potluckId;
 };
 
-const createUserPotluck = (potluckId, potluckInfo, uIds) => {
-  const userPotluck = {
-    isHost: false,
-    isNew: true,
-    eventDate: potluckInfo.eventDate,
-    title: potluckInfo.title,
-    description: potluckInfo.description,
-  };
+const processSignUpEmailInvites = (userId, email) => {
+  const encodedEmail = btoa(email);
 
-  uIds.forEach((uId) => {
-    const updates = {
-      [`/userPotLucks/${uId}/${potluckId}`]: userPotluck,
-    };
+  return getPotluckInvites()
+    .then((potluckIdResult) => {
+      return Object.keys(potluckIdResult.val());
+    })
+    .then((result) => {
+      const promises = result
+        .map((potluckId) => {
+          // search potluckIds for contained email and return potluckId if true
+          return getPotluckInviteEmail(potluckId, encodedEmail);
+        });
 
-    return api.change(updates);
-  });
+      return Promise.all(promises).then((emailMatch) => { 
+        return emailMatch;
+      });
```

Comment:
It seems likes you could just to:

```js
return Promise.all(promises);
```

The function in your `.then` doesn't actually do anything, so you shouldn't need it.